### DISCUSSION TITLE
add a `valid` field on submission

### DISCUSSION THREAD

**Discussion by alishabasnet** [edmundhung/conform#390] [Ideas]
On server side, it would be better if a `valid` field is added such that when:
`submission.valid` is true, only value object exists but no errors object and when `submission.valid` is false, only error object exists no value object exists.

this would allow us to do so:

```js
if(submission.valid) {
   // we know there is always submission.data, so
   console.log(submission.data.id);
} else {
  console.log(submission.error.id)
}

**Comment by edmundhung** [comment]
We will be introducing `submission.status` in which you will find `submission.value` only if the status is `success` without `submission.error`. You can find all the other changes [here](https://github.com/edmundhung/conform/releases/tag/v1.0.0-rc.0).
